cocos/ui/UIPageView.cpp

```diff
@@ -230,11 +231,23 @@ float PageView::getPositionXByIndex(ssize_t idx)const
 {
     return (getContentSize().width * (idx-_curPageIdx));
 }
+    
+float PageView::getPositionYByIndex(ssize_t idx)const
+{
+    return (getContentSize().height * (idx-_curPageIdx));
+}
 
 void PageView::onSizeChanged()
 {
     Layout::onSizeChanged();
-    _rightBoundary = getContentSize().width;
+    if (_direction == Direction::HORIZONTAL)
+    {
+        _rightBoundary = getContentSize().width;
+    }
+    else
+    {
+        _rightBoundary = getContentSize().height;
```

Comment:
There are two boundaries here: leftBoundary and rightBoundary, what's your suggestion about the name for left/bottom boundary and right/up boundary?

I can't find a good name.